add ability to activate inline edit fields via keyboard

ulferts opened the pull request:
Allows tabbing on the inline edit field.

https://community.openproject.com/work_packages/23076/activity

I feel that one scenario is still not supported the way it should be. If the user presses the "ESC" key while being in an inline edit input, the input disappears and the text representation is displayed as desired. But the focus is not placed on the text so it becomes hard to reactivate the field as the user has to start the tabbing journey anew. 

I couldn't get this to work as the field can only be focused after it reappears.

ulferts committed:
add ability to activate inline edit fields via keyboard

TeatroIO commented:
I've prepared a stage to preview changes. <a href='http://ulferts-feature-tabbable-inline-edit-fields.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com/' target='_blank'>Open stage</a> or <a href='http://ulferts-feature-tabbable-inline-edit-fields.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com//teatro.log' target='_blank'>view logs</a>.

ulferts committed:
soft require edit field directive as not every wp.td directive is nested

ulferts committed:
set tabindex of -1 for non editable fields

That way they will not become tabbable and hence not seem to editable to the user
